src/chat.rs

```diff
@@ -3385,6 +3386,8 @@ pub(crate) async fn add_info_msg_with_cmd(
     text: &str,
     cmd: SystemMessage,
     timestamp: i64,
+    // Timestamp to show to the user, if different than `timestamp` (which is used for sorting):
+    uservisible_timestamp: Option<i64>,
```

Comment:
maybe calling the parameters closer to the database fields make things a bit clearer? so maybe `timestamp_sort` and `timestamp_sent_rcvd`?